modules/3d/src/pointcloud/io_obj.cpp

```diff
@@ -39,9 +40,17 @@ void ObjDecoder::readData(std::vector<Point3f> &points, std::vector<Point3f> &no
             continue;
         else if (key == "v")
         {
-            Point3f vertex;
+            Point3f vertex, color;
             ss >> vertex.x >> vertex.y >> vertex.z;
             points.push_back(vertex);
+            if (ss.rdbuf()->in_avail() != 0) {
```

Comment:
Obj coordinates are x, y, z, [w]. `ss.rdbuf()->in_avail()` may report the last optional w. Also we could have  x, y, z, w, r, g, b. W should be handled correctly.